tests/test_proxy_functional.py

```diff
@@ -1,14 +1,137 @@
 import asyncio
 import os
 import pathlib
+import ssl
 from unittest import mock
 
+import proxy
 import pytest
 from yarl import URL
 
 import aiohttp
 from aiohttp import web
 
+# @pytest.fixture
+# def secure_proxy_url(
+#     aiohttp_unused_port, tls_ca_certificate_pem_path, tls_ca_private_key_pem_path
+# ):
+#     """Return the an instance of a running secure proxy."""
+#     unused_port = str(aiohttp_unused_port())
+#     address = "127.0.0.1"
+#     input_args = [
+#         "--num-workers",
+#         "1",
+#         "--hostname",
+#         "127.0.0.1",
+#         "--port",
+#         unused_port,
+#         "--cert-file",
+#         tls_ca_certificate_pem_path,
+#         "--key-file",
+#         tls_ca_private_key_pem_path,
+#     ]
+#     proxy_url = URL.build(scheme="https", host=address, port=unused_port)
+#     with proxy.Proxy(input_args=input_args):
+#         yield proxy_url
```

Comment:
This is what I wanted to use originally, and what I think we ultimately should use. I made the section below as an alternate version that very closely follows what trustme does here https://github.com/python-trio/trustme/blob/master/trustme/_cli.py#L50-L86

Ultimately I'm not sure the version of the fixture below actually improved it any so I'm leaving both versions for @webknjaz to see.